### DISCUSSION TITLE
LAI simulation is large for grassland and planthydro cannot be used

### DISCUSSION THREAD

**Discussion by niuhanlin** [NGEET/fates#920] [General]
When I run FATES-SP mode, LAI appears cone shaped. And when I run fixed biogeography mode , the LAI is trapezoidal(It grows exponentially and then linearly).
My purpose is to run FATES( fixed biogeography mode), but the current model has some problems with LAI:

        1.The growth period and the decline period are too fast.

        2.LAI remained unchanged during the period of vigorous growth from July to September. 

       3.The LAI maximum is too large for the grass. According to the literature and MODIS satellite data, 3.5 in SP-mode is relatively normal.

![TLAI_in_maqu_fates_sp_spinup_200years clm2 h0 2012](https://user-images.githubusercontent.com/66841974/196129697-06a498b8-63b4-4068-a17b-d185ffd44f04.png)
![TLAI_in_maqu_fates_sp_spinup_200years_again clm2 h](https://user-images.githubusercontent.com/66841974/196129777-8fd0a209-b11c-4818-816f-6fa941afe5ab.png)

So I think it's important to modify the parameters of the allometric assignment curve for grasslands.
Does anyone have any suggestions for that?   Is there a realistic parameter for the grass?

Another issue is that currently the program will report an error when use_fates_planthydro =.true. is set.
[cesm.log.txt](https://github.com/NGEET/fates/files/9798526/cesm.log.txt)
[lnd.log.txt](https://github.com/NGEET/fates/files/9798528/lnd.log.txt)

using version: [sci.1.59.7_api.24.1.0]

**Comment by rosiealice** [comment]
I agree part of the bias with lai may come from parametric issues. ,y suspicion here is that the lai is reaching the 'cap' imposed by the lai trimming mechanism, and losing any leaf layers in negative carbon balance. The reduce that cap one should either decrease productivity or increase turnover. One simple thing would be to modify root and /or lead turnover rates and see what impact this has. 

To reduce the rate of 'ramp up' is more complicated but probably results from npp being too high. You could look at the impact of vcmax to check whether that reduces the leaf biomass increase. 

Senescence is more complex and probably is more related to the phenology model not capturing the processes of leaf aging and late summer senescence across the riversity of phenological strategies we aren't capturing... 

The SP output should be the same as the surface dataset inputs, with all the caveats of spatial averaging and satellite data products.

Hope that helps! 
Rosie

**Reply by niuhanlin** [reply]
@rosiealice 
Thank you for your advice.
According to your suggestion, I reduced the VCMAX artificially (without any meaning). I just want to see whether the reduction of VCMAX will reduce LAI. The answer is yes, but only from 4.5 down to 4.4.

As for the reduction of npp and increase of turnover rate you mentioned, I am not very clear about their specific positions. Could you please point them out?

I think if the above problems can be solved, we will continue to discuss how to improve LAI changes in the fading period.

**Comment by rgknox** [comment]
Here is a log of questions and answers on this topic during the FATES model meeting October 27th:

```
From Hanlin:

1. I am using fixed_biogeog mode, but I cannot use planthydro in this
case. I tried to change fates_hydro_solver from 1 to 2, but the result
was the same error.
ERROR content: ERROR: One or more of the output from CLM to the coupler are NaN

Ryan:  
(my suspicion is that there are nan’s probably generated by the fates hydro solver, 
and these nans generate more nans in other places, and the coupler is the first place 
that is checking for nans, even though they already exist)  If you run in debug mode, 
I bet you will find a math error occurring at a different location. Does the model run 
when planthydro is turned on? The module itself should be compatible with fixed 
biogeography and no-comp.  I haven't considered using it with SP, and would avoid 
that for now.  A key issue that still persists with hydro, is that the land-model’s ET 
solution could demand more integrated transpiration flux from a fates cohort over 
one timestep, than what is currently in the leaf, which is impossible for the solver to
 handle.  The workaround is to use Chonggang’s PR.  But I’m not sure if this is your problem.

./xmlchange DEBUG=True

Chonggang’s PR: 851 https://github.com/NGEET/fates/pull/851

Yilin:  Is this on initialization, or rather, when in the simulation did this trigger? 
Is this plot-scale, or is this a regional/global?


2. I have found that LAI has a great influence on GPP. At present, LAI
changes too fast in the growth and decline stages, so I want to
improve GPP by changing the allometric growth curve.

When I changed the parameter fates_allom_d2bl1 from 0.07 to 0.048,
the vegetation died directly in the later stage, I don't understand
what went wrong?

Ryan:  
Regarding growth and decline.  Maybe also look at the leaf flushing algorithm?  
Currently, when temperature or moisture signals that leaves can flush, there is no 
limitation on how fast they can get back up to last year’s target (defined by “dbh” 
and allometry). If there is carbon from either NPP or large stores, it will grow leaf
 until it hits the target.  I’m imagining a new cohort variable like “flush_fraction”, 
that coordinates with the bleaf() allometry scheme, and sets a reduced leaf target.
The leaf target can grow from 0 to 1 over a more reasonable window.

Here is where we set the amount of carbon that is flushed (from storage)  immediately on the trigger:
https://github.com/NGEET/fates/blob/sci.1.59.7_api.24.1.0/biogeochem/EDPhysiologyMod.F90#L1132-L1133
From the default file, we flush 50% of the leaf immediately upon the trigger:
https://github.com/NGEET/fates/blob/sci.1.59.7_api.24.1.0/parameter_files/fates_params_default.cdl#L1248-L1249
Here is the subroutine for bleaf(), which could potentially accept a new argument: 
https://github.com/NGEET/fates/blob/sci.1.59.7_api.24.1.0/biogeochem/FatesAllometryMod.F90#L519-L553

Regarding vegetation death: 
https://github.com/NGEET/fates/blob/sci.1.59.7_api.24.1.0/biogeochem/FatesAllometryMod.F90#L409-L439
When d2bl1 is decreased, you get less total leaf biomass for a plant of a given size. If it doesn’t have enough leaf, it won’t necessarily generate more photosynthate than carbon used in respiration.




```

**Comment by niuhanlin** [comment]
@rgknox 
I agree with you. I also tested a lot, and finally found that it was the problem of maintaining continuous linear growth of LAI in the decline period.
Maybe we should start with a declination function like the one in the CLM instead of having it end up at a certain point.
